src/test/java/teammates/architecture/ArchitectureTest.java

```diff
@@ -351,7 +351,13 @@ public boolean apply(JavaClass input) {
                     }
                 })
                 .orShould().accessClassesThat().resideInAPackage(includeSubpackages(LOGIC_PACKAGE))
-                .orShould().accessClassesThat().resideInAPackage(includeSubpackages(UI_PACKAGE))
+                .orShould().accessClassesThat(new DescribedPredicate<>("") {
+                    @Override
+                    public boolean apply(JavaClass input) {
+                        return input.getPackageName().startsWith(UI_PACKAGE)
+                                && !input.getPackageName().startsWith(UI_OUTPUT_PACKAGE);
+                    }
+                })
```

Comment:
wld like to point this out, here we are letting e2e tests access the output, as its needed to do assertions (the old way with datastore was to convert the Data to the actual DTO, but that cannot be done now as the Data does not actually contain all the required fields to create the entity (this is due to SQL being relational, e.g. return FeedbackResponseData, we do not actually have the full course to create the entity)

see AbstractBackDoor.java for examples of how this Data -> Entity is done in datastore, and i think its not possible for our new SQL setup